agent/xds/proxysupport/proxysupport.go

```diff
@@ -12,3 +14,28 @@ var EnvoyVersions = []string{
 	"1.22.5",
 	"1.21.5",
 }
+
+// UnsupportedEnvoyVersions lists any unsupported Envoy versions (mainly minor versions) that fall
+// within the range of EnvoyVersions above.
+// For example, if patch 1.21.3 (patch 3) had a breaking change, and was not supported
+// even though 1.21 is a supported major release, you would then add 1.21.3 to this list.
+// This list will be empty in most cases.
+//
+// see: https://www.consul.io/docs/connect/proxies/envoy#supported-versions
+var UnsupportedEnvoyVersions = []string{}
+
+// GetMaxEnvoyMajorVersion grabs the first value in EnvoyVersions and strips the patch number off in order
+// to return the maximum supported Envoy Major version
+// For example, if the input string is "1.14.1", the function would return "1.14".
+func GetMaxEnvoyMajorVersion() string {
+	s := strings.Split(EnvoyVersions[0], ".")
+	return s[0] + "." + s[1]
+}
+
+// GetMinEnvoyMajorVersion grabs the last value in EnvoyVersions and strips the patch number off in order
+// to return the minimum supported Envoy Major version
+// For example, if the input string is "1.12.1", the function would return "1.12".
+func GetMinEnvoyMajorVersion() string {
+	s := strings.Split(EnvoyVersions[len(EnvoyVersions)-1], ".")
```

Comment:
Do we need to handle the unlikely case that the `EnvoyVersions` array is misconfigured (e.g., empty list, or a value that doesn't contain a `.`).? Maybe not since this is a hard-coded internal array, but my brain spotted this as a hypothetical, potential out-of-bounds access issue (both on array `s` and array `EnvoyVersions`).

Reply:
Same applies to the get max version func

Reply:
I guess if we don't have `.` we have bigger problems, we wouldn't be able to parse the string out of `envoy --version` as well.